frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx

```diff
@@ -1945,6 +1948,37 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
                 }
             },
         ],
+        authorizedDomains: [
+            (s) => [s.authorizedUrls],
+            (authorizedUrls) => {
+                // There are a couple problems with the raw `authorizedUrls` which we need to fix here:
+                // - They are URLs, we want domains
+                // - There might be duplicates, so clean them up
+                // - There might be duplicates across http/https, so clean them up
+
+                // First create URL objects and group them by hostname+port
+                const urlsByDomain = new Map<string, URL[]>()
+
+                for (const urlStr of authorizedUrls) {
+                    try {
+                        const url = new URL(urlStr)
+                        const key = url.host // hostname + port if present
+                        if (!urlsByDomain.has(key)) {
+                            urlsByDomain.set(key, [])
+                        }
+                        urlsByDomain.get(key)!.push(url)
+                    } catch (e) {
+                        // Silently skip URLs that can't be parsed
+                    }
+                }
+
+                // For each domain, prefer https over http
+                return Array.from(urlsByDomain.values()).map((urls) => {
+                    const preferredUrl = urls.find((url) => url.protocol === 'https:') ?? urls[0]
+                    return preferredUrl.origin
+                })
```

Comment:
Cool! This should also fix problem with some urls showing up with data like `http://posthog.com/` but not `https://posthog.com/` like @andyvan-ph mentioned.

Reply:
Yeah, that was a case of us having misconfigured our own URLs in our project, but it should help with HTTP AND HTTPS existing at the same time.